pkg/nodeagent/certificate.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeagent
+
+import (
+	"context"
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	nodeagentv1alpha1 "github.com/gardener/gardener/pkg/nodeagent/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/csr"
+)
+
+const nodeAgentCSRPrefix = "node-agent-csr-"
+
+// RequestAndStoreKubeconfig requests a certificate via CSR and stores the resulting kubeconfig on the disk.
+func RequestAndStoreKubeconfig(ctx context.Context, log logr.Logger, fs afero.Afero, config *rest.Config, machineName string) error {
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	certificateSubject := &pkix.Name{
+		Organization: []string{v1beta1constants.NodeAgentsGroup},
+		CommonName:   v1beta1constants.NodeAgentUserNamePrefix + machineName,
```

Comment:
What was the reason to use the machine name and not the Node name? Because with the machine name it is easier later on in the authorizer webhook to find the corresponding Machine?